### PR TITLE
[FLINK-12403][doc-zh]Translate 'How to use logging' page into Chinese

### PR DESCRIPTION
**Pull request by hapihu**
Translate 'How to use logging' page into Chinese

### CODE REVIEW

**Comment by wuchong** [comment]
cc @RocMarshal , could you help to review this?

**Comment by RocMarshal** [comment]
Thanks for the contribution of @hapihu .   Thanks for the trust. @wuchong  I'll check it ASAP.

**Comment by RocMarshal** [comment]
@hapihu Would you mind checking the this page [in the content.zh] with the latest page [in the content]? I notice that there seems to be some differences(not language)  in between this page and the page in english version.

**Comment by hapihu** [comment]
> @hapihu Would you mind checking the this page [in the content.zh] with the latest page [in the content]? I notice that there seems to be some differences(not language) between this page and the page in english version.

@RocMarshal ，Ok. Received. Let me check first. Thank you~

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 1]
```diff
@@ -27,86 +27,102 @@ under the License.
 
 # 如何使用日志记录
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:28]:
```suggestion
<a name="how-to-use-logging"></a>

# 如何使用日志记录
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 2]
```diff
@@ -27,86 +27,102 @@ under the License.
...
 
-Flink 中的日志记录是使用 slf4j 日志接口实现的。使用 log4j2 作为底层日志框架。我们也支持了 logback 日志配置，只要将其配置文件作为参数传递给 JVM 即可。愿意使用 logback 而不是 log4j2 的用户只需排除 log4j2 的依赖（或从 lib/ 文件夹中删除它）即可。
+所有 Flink 进程都会创建一个文本格式的日志文件，其中包含该进程中发生的各种事件的信息。
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:31]:
```suggestion
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 3]
```diff
@@ -27,86 +27,102 @@ under the License.
...
 
+这些日志提供了深入了解 Flink 内部工作的途径，同时可以用来输出检测出的问题（以 WARN/ERROR 消息的形式），还可以辅助调试问题。
 
+日志文件可以通过 Job-/TaskManager 对应的 WebUI 页面访问。所使用的 [Resource Provider]({{< ref "docs/deployment/overview" >}}) （如 YARN）可能会提供额外的访问方式来访问日志。
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:34]:
```suggestion
日志文件可以通过 Job-/TaskManager 对应的 WebUI 页面访问。所使用的 [Resource Provider]({{< ref "docs/deployment/overview" >}})（如 YARN）可能会提供额外的访问方式来访问日志。
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 4]
```diff
@@ -27,86 +27,102 @@ under the License.
...
 
-<a name="configuring-log4j2"></a>
+Flink 中的日志记录是使用 [SLF4J](http://www.slf4j.org/) 日志接口实现的。这允许你不需要修改 Flink 的源代码就可以使用任何支持 SLF4J 的日志框架。
+
+默认情况下，使用 [Log4j 2](https://logging.apache.org/log4j/2.x/index.html) 作为底层日志框架。
+
+<a name="configuring-log4j2"> </a>
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:40]:
```suggestion
<a name="configuring-log4j-2"></a>
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 5]
```diff
@@ -27,86 +27,102 @@ under the License.
...
 
 ## 配置 Log4j2
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:42]:
```suggestion
## 配置 Log4j 2
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 6]
```diff
@@ -27,86 +27,102 @@ under the License.
...
 
-Log4j2 是使用配置文件指定的。在 Flink 的使用中，该文件通常命名为 `log4j.properties`。我们使用 `-Dlog4j.configurationFile=` 参数将该文件的文件名和位置传递给 JVM。
+Log4j2 是通过 property 配置文件进行配置的。
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:44]:
```suggestion
Log4j 2 是通过 property 配置文件进行配置的。
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 7]
```diff
@@ -27,86 +27,102 @@ under the License.
...
 
-Flink 附带以下默认日志配置文件：
+Flink 发行版在 `conf` 目录中附带了以下 log4j 配置文件，如果启用了 Log4j 2，则会自动使用如下文件：
 
-- `log4j-cli.properties`：由 Flink 命令行客户端使用（例如 `flink run`）（不包括在集群上执行的代码）
-- `log4j-session.properties`：Flink 命令行客户端在启动 YARN 或 Kubernetes session 时使用（`yarn-session.sh`，`kubernetes-session.sh`）
-- `log4j.properties`：作为 JobManager/TaskManager 日志配置使用（standalone 和 YARN 两种模式下皆使用）
+- `log4j-cli.properties`：Flink 命令行端使用（例如 `flink run`）；
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:48]:
```suggestion
- `log4j-cli.properties`：Flink 命令行使用（例如 `flink run`）；
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 8]
```diff
@@ -27,86 +27,102 @@ under the License.
...
+- `log4j-session.properties`：Flink 命令行端在启动基于 Yarn 或 Kubernetes 的 Session 集群时使用（例如 `yarn-session.sh` 或 `kubernetes-session.sh`）；
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:49]:
```suggestion
- `log4j-session.properties`：Flink 命令行在启动基于 Kubernetes/Yarn 的 Session 集群时使用（例如 `kubernetes-session.sh`/`yarn-session.sh`）；
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 9]
```diff
@@ -27,86 +27,102 @@ under the License.
...
+- `log4j-console.properties`：JobManager/Taskmanagers 在前端模式运行时使用（例如 Kubernetes）；
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:50]:
```suggestion
- `log4j-console.properties`：Job-/TaskManagers 在前台模式运行时使用（例如 Kubernetes）；
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 10]
```diff
@@ -27,86 +27,102 @@ under the License.
...
+- `log4j.properties`： JobManager/TaskManager 默认使用的日志配置。
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:51]:
```suggestion
- `log4j.properties`： Job-/TaskManagers 默认使用的日志配置。
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 11]
```diff
@@ -27,86 +27,102 @@ under the License.
...
 
-<a name="compatibility-with-log4j1"></a>
+Log4j 会定期扫描这些文件的变更，并在必要时调整日志记录行为。默认情况下30秒检查一次，监测间隔可以通过 Log4j 配置文件的 `monitorInterval` 配置项进行设置。
 
-### 与 Log4j1 的兼容性
+<a name="compatibility-with-log4j1"> </a>
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:55]:
```suggestion
<a name="compatibility-with-log4j-1"></a>
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 12]
```diff
@@ -27,86 +27,102 @@ under the License.
...
 
-Flink 附带了 [Log4j API bridge](https://logging.apache.org/log4j/log4j-2.2/log4j-1.2-api/index.html)，使得现有作业能够继续使用 log4j1 的接口。
+### 适配 Log4j1
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:57]:
```suggestion
### 与 Log4j 1 的兼容
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 13]
```diff
@@ -27,86 +27,102 @@ under the License.
...
 
-如果你有基于 Log4j 的自定义配置文件或代码，请查看官方 Log4j [兼容性](https://logging.apache.org/log4j/2.x/manual/compatibility.html)和[迁移](https://logging.apache.org/log4j/2.x/manual/migration.html)指南。
+Flink 附带了 [Log4j API bridge](https://logging.apache.org/log4j/log4j-2.2/log4j-1.2-api/index.html) 相关的依赖，使当前基于 Log4j1 开发的应用程序可以继续正常运行。
 
-<a name="configuring-log4j1"></a>
+如果有基于 Log4j 的自定义配置文件或代码，请查看官方 Log4j [适配](https://logging.apache.org/log4j/2.x/manual/compatibility.html)和[迁移](https://logging.apache.org/log4j/2.x/manual/migration.html)指南。
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:61]:
```suggestion
如果有基于 Log4j 1 的自定义配置文件或代码，请查看官方 Log4j [兼容](https://logging.apache.org/log4j/2.x/manual/compatibility.html)和[迁移](https://logging.apache.org/log4j/2.x/manual/migration.html)指南。
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 14]
```diff
@@ -27,86 +27,102 @@ under the License.
...
+
+<a name="configuring-log4j1"> </a>
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:63]:
```suggestion
<a name="configuring-log4j1"></a>
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 15]
```diff
@@ -27,86 +27,102 @@ under the License.
...
 
-要将 Flink 与 Log4j1 一起使用，必须确保：
-- Classpath 中不存在 `org.apache.logging.log4j:log4j-core`，`org.apache.logging.log4j:log4j-slf4j-impl` 和 `org.apache.logging.log4j:log4j-1.2-api`，
-- 且 Classpath 中存在 `log4j:log4j`，`org.slf4j:slf4j-log4j12`，`org.apache.logging.log4j:log4j-to-slf4j` 和 `org.apache.logging.log4j:log4j-api`。
+要将 Flink 与 [Log4j 1](https://logging.apache.org/log4j/1.2/) 一起使用，必须确保：
+- Classpath 中不存在 `org.apache.logging.log4j:log4j-core`、`org.apache.logging.log4j:log4j-slf4j-impl` 和 `org.apache.logging.log4j:log4j-1.2-api`；
+- 且 Classpath 中存在 `log4j:log4j`、`org.slf4j:slf4j-log4j12`、`org.apache.logging.log4j:log4j-to-slf4j` 和 `org.apache.logging.log4j:log4j-api`。
 
-在 IDE 中使用 log4j1，你必须在 pom 文件中使用上述 `Classpath 中存在的 jars` 依赖项替换 `Classpath 中不存在的 jars` 依赖项，并尽可能在传递依赖于 `Classpath 中不存在的 jars` 的依赖项上添加排除 `Classpath 中不存在的 jars` 配置。
+在 IDE 中使用 Log4j1 ，必须在 pom 文件中使用上述 Classpath 中应该存在的 jars 依赖项来替换  Classpath 中不应该存在的 jars 依赖项，并尽可能的排除那些传递依赖于 Classpath 中不存在 jars 的依赖项。
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:71]:
```suggestion
如果在 IDE 中使用 Log4j 1，则必须在 pom 文件中使用上述 Classpath 中应该存在的 jars 依赖项来替换 Classpath 中不应该存在的 jars 依赖项，并尽可能的排除那些传递依赖于 Classpath 中不存在 jars 的依赖项。
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 16]
```diff
@@ -27,86 +27,102 @@ under the License.
...
 
-对于 Flink 发行版，这意味着你必须
-- 从 `lib` 目录中移除 `log4j-core`，`log4j-slf4j-impl` 和 `log4j-1.2-api` jars，
-- 向 `lib` 目录中添加 `log4j`，`slf4j-log4j12` 和 `log4j-to-slf4j` jars，
-- 用兼容的 Log4j1 版本替换 `conf` 目录中的所有 log4j 配置文件。
+对于 Flink 发行版，这意味着你必须：
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:73]:
```suggestion
对于 Flink 发行版，这意味着你必须
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 17]
```diff
@@ -27,86 +27,102 @@ under the License.
...
+- 从 `lib` 目录中移除 `log4j-core`，`log4j-slf4j-impl` 和 `log4j-1.2-api` jars；
+- 往 `lib` 目录中添加 `log4j`，`slf4j-log4j12` 和 `log4j-to-slf4j` jars；
+- 用适配的 Log4j1 版本替换 `conf` 目录中的所有 log4j 配置文件。
 
-<a name="configuring-logback"></a>
+<a name="configuring-logback"> </a>
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:78]:
```suggestion
<a name="configuring-logback"></a>
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 18]
```diff
@@ -27,86 +27,102 @@ under the License.
...
-            <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{60} %X{sourceThread} - %msg%n</pattern>
-        </encoder>
-    </appender>
-
-    <root level="INFO">
-        <appender-ref ref="file"/>
-    </root>
-</configuration>
-```
+要将 Flink 与 [logback](https://logback.qos.ch/) 一起使用，必须确保：
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:83]:
```suggestion
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 19]
```diff
@@ -27,86 +27,102 @@ under the License.
...
 
-例如，为了控制 `org.apache.flink.runtime.jobgraph.JobGraph` 的日志记录级别，必须将以下行添加到配置文件中。
+- Classpath 中不存在 `org.apache.logging.log4j:log4j-slf4j-impl`；
+- Classpath 中存在 `ch.qos.logback:logback-core` 和 `ch.qos.logback:logback-classic`。
 
-```xml
-<logger name="org.apache.flink.runtime.jobgraph.JobGraph" level="DEBUG"/>
-```
+在 IDE 中使用 logback ，必须在 pom 文件中使用上述 Classpath 中应该存在的 jars 依赖项来替换  Classpath 中不应该存在的 jars 依赖项，并尽可能的排除那些传递依赖于 Classpath 中不存在 jars 的依赖项。
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:87]:
```suggestion
如果在 IDE 中使用 logback，则必须在 pom 文件中使用上述 Classpath 中应该存在的 jars 依赖项来替换 Classpath 中不应该存在的 jars 依赖项，并尽可能的排除那些传递依赖于 Classpath 中不存在 jars 的依赖项。
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 20]
```diff
@@ -27,86 +27,102 @@ under the License.
...
+
+对于 Flink 发行版，这意味着你必须：
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:89]:
```suggestion
对于 Flink 发行版，这意味着你必须
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 21]
```diff
@@ -27,86 +27,102 @@ under the License.
...
+
+- 从 `lib` 目录中移除 `log4j-slf4j-impl`  jars；
+- 向 `lib` 目录中添加 `logback-core` 和 `logback-classic` jars。
+
+Flink 发行版在 `conf` 目录中附带了以下 logback 配置文件，如果启用了 logback，则会自动使用这些文件：
 
-有关配置日志的更多信息，请参见 [LOGback 手册](http://logback.qos.ch/manual/configuration.html)。
+- `logback-session.properties`: Flink 命令行端在启动基于 Yarn 或 Kubernetes 的 Session 集群时使用（例如 `yarn-session.sh` 或 `kubernetes-session.sh`）；
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:96]:
```suggestion
- `logback-session.properties`: Flink 命令行在启动基于 Kubernetes/Yarn 的 Session 集群时使用（例如 `kubernetes-session.sh`/`yarn-session.sh`）；
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 22]
```diff
@@ -27,86 +27,102 @@ under the License.
...
+- `logback-console.properties`：JobManager/Taskmanagers 在前端模式运行时使用（例如 Kubernetes）；
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:97]:
```suggestion
- `logback-console.properties`：Job-/TaskManagers 在前台模式运行时使用（例如 Kubernetes）；
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 23]
```diff
@@ -27,86 +27,102 @@ under the License.
...
+- `logback.xml`: 命令行端和 JobManager/TaskManager 默认使用的日志配置。
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:98]:
```suggestion
- `logback.xml`: 命令行和 Job-/TaskManager 默认使用的日志配置。
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 24]
```diff
@@ -27,86 +27,102 @@ under the License.
...
 
-<a name="best-practices-for-developers"></a>
+{{< hint warning >}}
+
+Logback 1.3+ 需要 SLF4J 2，目前不支持。
+
+{{< /hint >}}
+
+
+<a name="best-practices-for-developers"> </a>
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:107]:
```suggestion
<a name="best-practices-for-developers"></a>
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 25]
```diff
@@ -27,86 +27,102 @@ under the License.
...
 
 ## 开发人员的最佳实践
 
-Slf4j 的 loggers 通过调用 `LoggerFactory` 的 `getLogger()` 方法创建
+通过将相应 Class 的类名称作为参数来调用 `org.slf4j.LoggerFactory#LoggerFactory.getLogger` 方法可以创建一个 Slf4j 的 logger。
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:111]:
```suggestion
通过将相应 Class 的类型对象作为参数调用 `org.slf4j.LoggerFactory#LoggerFactory.getLogger` 方法可以创建一个 SLF4J 的 logger。
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 26]
```diff
@@ -27,86 +27,102 @@ under the License.
...
+
+强烈建议将 logger 字段设置为 `private static final` 类型。
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:113]:
```suggestion
强烈建议将 logger 字段设置为 `private static final` 修饰的类型。
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 27]
```diff
@@ -27,86 +27,102 @@ under the License.
...
+	private static final Logger LOG = LoggerFactory.getLogger(Foobar.class);
 
-Logger LOG = LoggerFactory.getLogger(Foobar.class)
+	public static void main(String[] args) {
+		LOG.info("Hello world!");
+	}
+}
 ```
 
 为了最大限度地利用 slf4j，建议使用其占位符机制。使用占位符可以避免不必要的字符串构造，以防日志级别设置得太高而不会记录消息。占位符的语法如下：
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:128]:
```suggestion
为了最大限度地利用 SLF4J，建议使用其占位符机制。使用占位符可以在日志级别设置得太高而不会记录消息的情况下避免不必要的字符串构造。

占位符的语法如下：
```

**Comment by hapihu** [comment]
Hello, @RocMarshal 
Excuse me for taking up your time.
I have finished the modification according to your suggestion.
Would you like to review it for me again.
Thank you very much!

**Comment by RocMarshal** [comment]
@hapihu Hi, I notice that there are still some comments without your response. Maybe you could click the 'Load More...' to get more details. Thank you.

**Comment by hapihu** [comment]
@RocMarshal ，hi，I'm sorry I didn't notice all the comments before.
Now, I have finished the modification according to your suggestion.
Would you like to review it for me again.
Thank you very much!

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 28]
```diff
@@ -25,91 +25,109 @@ specific language governing permissions and limitations
...
+- Classpath 中存在 `ch.qos.logback:logback-core` 和 `ch.qos.logback:logback-classic`。
 
-例如，为了控制 `org.apache.flink.runtime.jobgraph.JobGraph` 的日志记录级别，必须将以下行添加到配置文件中。
+如果在 IDE 中使用 logback，则必须在 pom 文件中使用上述 Classpath 中应该存在的 jars 依赖项来替换 Classpath 中不应该存在的 jars 依赖项，并尽可能的排除那些传递依赖于 Classpath 中不存在 jars 的依赖项。
 
-```xml
-<logger name="org.apache.flink.runtime.jobgraph.JobGraph" level="DEBUG"/>
-```
+对于 Flink 发行版，这意味着你必须
+
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:90]:
```suggestion
```

File: docs/content.zh/docs/deployment/advanced/logging.md [hunk 29]
```diff
@@ -25,91 +25,109 @@ specific language governing permissions and limitations
...
+- 从 `lib` 目录中移除 `log4j-slf4j-impl`  jars；
+- 向 `lib` 目录中添加 `logback-core` 和 `logback-classic` jars。
+
+Flink 发行版在 `conf` 目录中附带了以下 logback 配置文件，如果启用了 logback，则会自动使用这些文件：
+
```
Comment (RocMarshal) [inline docs/content.zh/docs/deployment/advanced/logging.md:95]:
```suggestion
```

**Review by RocMarshal — APPROVED** [review]
Good job @hapihu . LGTM +1.  Now, ping @wuchong .

**Comment by wuchong** [comment]
Thanks for the reviewing @RocMarshal  and the contribution @hapihu . 
Merging...

**Pull request merged**